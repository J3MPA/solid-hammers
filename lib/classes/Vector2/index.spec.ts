import { Vector2 } from '.'

describe('Vector2', () => {
  describe('Sad path', () => {
    it('should trow TypeError if Vector2 is not instantiated with x as Number', () => {
      expect(() => new Vector2('some' as any, 2)).toThrow(TypeError)
      expect(() => new Vector2('some' as any, 2)).toThrowErrorMatchingSnapshot()
    })
    it('should trow TypeError if Vector2 is not instantiated with y as Number', () => {
      expect(() => new Vector2(2, 'some' as any)).toThrow(TypeError)
      expect(() => new Vector2(2, 'some' as any)).toThrowErrorMatchingSnapshot()
    })
  })
})
