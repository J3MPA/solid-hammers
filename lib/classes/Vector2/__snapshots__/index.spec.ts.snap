// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vector2 constructor Sad path should throw TypeError if Vector2 is instantiated with x as a non real number 1`] = `"Point must be a real number"`;

exports[`Vector2 constructor Sad path should throw TypeError if Vector2 is instantiated with x as a non real number 2`] = `"Point must be a real number"`;

exports[`Vector2 constructor Sad path should throw TypeError if Vector2 is instantiated with x as a non real number 3`] = `"Point must be a real number"`;

exports[`Vector2 constructor Sad path should throw TypeError if Vector2 is instantiated with x as a non real number 4`] = `"Point must be a real number"`;

exports[`Vector2 constructor Sad path should throw TypeError if Vector2 is instantiated with x as a non real number 5`] = `"Point must be a real number"`;

exports[`Vector2 constructor Sad path should throw TypeError if Vector2 is instantiated with x as a non real number 6`] = `"Point must be a real number"`;

exports[`Vector2 constructor Sad path should throw TypeError if Vector2 is instantiated with x as a non real number 7`] = `"Point must be a real number"`;

exports[`Vector2 constructor Sad path should throw TypeError if Vector2 is instantiated with y as a non real number 1`] = `"Point must be a real number"`;

exports[`Vector2 constructor Sad path should throw TypeError if Vector2 is instantiated with y as a non real number 2`] = `"Point must be a real number"`;

exports[`Vector2 constructor Sad path should throw TypeError if Vector2 is instantiated with y as a non real number 3`] = `"Point must be a real number"`;

exports[`Vector2 constructor Sad path should throw TypeError if Vector2 is instantiated with y as a non real number 4`] = `"Point must be a real number"`;

exports[`Vector2 constructor Sad path should throw TypeError if Vector2 is instantiated with y as a non real number 5`] = `"Point must be a real number"`;

exports[`Vector2 constructor Sad path should throw TypeError if Vector2 is instantiated with y as a non real number 6`] = `"Point must be a real number"`;

exports[`Vector2 constructor Sad path should throw TypeError if Vector2 is instantiated with y as a non real number 7`] = `"Point must be a real number"`;

exports[`Vector2 prototype.scale Sad path should throw TypeError if scalar is not a real number 1`] = `"Scalar must be a real number"`;

exports[`Vector2 prototype.scale Sad path should throw TypeError if scalar is not a real number 2`] = `"Scalar must be a real number"`;

exports[`Vector2 prototype.scale Sad path should throw TypeError if scalar is not a real number 3`] = `"Scalar must be a real number"`;

exports[`Vector2 prototype.scale Sad path should throw TypeError if scalar is not a real number 4`] = `"Scalar must be a real number"`;

exports[`Vector2 prototype.scale Sad path should throw TypeError if scalar is not a real number 5`] = `"Scalar must be a real number"`;

exports[`Vector2 prototype.scale Sad path should throw TypeError if scalar is not a real number 6`] = `"Scalar must be a real number"`;

exports[`Vector2 prototype.scale Sad path should throw TypeError if scalar is not a real number 7`] = `"Scalar must be a real number"`;

exports[`Vector2 static fromMagnitute Sad path should throw TypeError if magnitute is not a real number 1`] = `"Magnitute must be a real number"`;

exports[`Vector2 static fromMagnitute Sad path should throw TypeError if magnitute is not a real number 2`] = `"Magnitute must be a real number"`;

exports[`Vector2 static fromMagnitute Sad path should throw TypeError if magnitute is not a real number 3`] = `"Magnitute must be a real number"`;

exports[`Vector2 static fromMagnitute Sad path should throw TypeError if magnitute is not a real number 4`] = `"Magnitute must be a real number"`;

exports[`Vector2 static fromMagnitute Sad path should throw TypeError if magnitute is not a real number 5`] = `"Magnitute must be a real number"`;

exports[`Vector2 static fromMagnitute Sad path should throw TypeError if magnitute is not a real number 6`] = `"Magnitute must be a real number"`;

exports[`Vector2 static fromMagnitute Sad path should throw TypeError if magnitute is not a real number 7`] = `"Magnitute must be a real number"`;

exports[`Vector2 static scale Sad path should throw TypeError if scalar is not a real number 1`] = `"Scalar must be a real number"`;

exports[`Vector2 static scale Sad path should throw TypeError if scalar is not a real number 2`] = `"Scalar must be a real number"`;

exports[`Vector2 static scale Sad path should throw TypeError if scalar is not a real number 3`] = `"Scalar must be a real number"`;

exports[`Vector2 static scale Sad path should throw TypeError if scalar is not a real number 4`] = `"Scalar must be a real number"`;

exports[`Vector2 static scale Sad path should throw TypeError if scalar is not a real number 5`] = `"Scalar must be a real number"`;

exports[`Vector2 static scale Sad path should throw TypeError if scalar is not a real number 6`] = `"Scalar must be a real number"`;

exports[`Vector2 static scale Sad path should throw TypeError if scalar is not a real number 7`] = `"Scalar must be a real number"`;
